### PR TITLE
22588 - Revert partner disbursements, fix migration processes, update versions, prod PREP

### PR DESCRIPTION
**Pull request by seeker25**
https://github.com/bcgov/entity/issues/22588

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: jobs/payment-jobs/tasks/ejv_payment_task.py [hunk 1]
```diff
@@ -227,18 +227,16 @@ def _get_account_ids_for_payment(cls, batch_type) -> List[int]:
...
             .filter(DistributionCodeModel.stop_ejv.is_(False) | DistributionCodeModel.stop_ejv.is_(None)) \
-            .filter(DistributionCodeModel.account_id.isnot(None))
-
-        if batch_type == 'GA':
-            # Rule for GA. Credit is 112 and debit is 112. For BCREG client code is 112
-            account_ids: List[int] = query.filter(DistributionCodeModel.client == bc_reg_client_code)
-        else:
-            # Rule for GI. Credit is 112 and debit is not 112. For BCREG client code is 112
-            account_ids: List[int] = query.filter(DistributionCodeModel.client != bc_reg_client_code)
-        return db.session.scalars(account_ids).all()
```
Comment (seeker25) [inline jobs/payment-jobs/tasks/ejv_payment_task.py:241]: This doesn't work

File: jobs/payment-jobs/tasks/ejv_partner_distribution_task.py [hunk 2]
```diff
@@ -234,39 +195,20 @@ def _create_ejv_file_for_partner(cls, batch_type: str):  # pylint:disable=too-ma
...
-                    refund_partial.disbursement_status_code = DisbursementStatus.UPLOADED.value
-
-            # Create ejv invoice/partial_refund link records and set invoice status
             sequence = 1
-            sequence = cls._create_ejv_link(invoices, ejv_header_model, sequence, EJVLinkType.INVOICE.value)
-            cls._create_ejv_link(refund_partial_items, ejv_header_model, sequence, EJVLinkType.REFUND.value)
+            # Create ejv invoice link records and set invoice status
+            for inv in invoices:
+                # Create Ejv file link and flush
+                link_model = EjvInvoiceLinkModel(invoice_id=inv.id,
```
Comment (seeker25) [inline jobs/payment-jobs/tasks/ejv_partner_distribution_task.py:202]: probably need to fix these to EJV_LINKS or whatever the new name is

File: pay-api/src/pay_api/services/base_payment_system.py [hunk 3]
```diff
@@ -150,7 +151,8 @@ def ensure_no_payment_blockers(self, payment_account: PaymentAccount) -> None:
             # Note NSF (Account Unlocking) is paid using DIRECT_PAY - CC flow, not PAD.
             current_app.logger.warning(f'Account {payment_account.id} is frozen, rejecting invoice creation')
             raise BusinessException(Error.PAD_CURRENTLY_NSF)
-        if Invoice.has_overdue_invoices(payment_account.id):
+        if flags.is_on('enable-eft-payment-method', default=False) and \
+            Invoice.has_overdue_invoices(payment_account.id):
```
Comment (seeker25) [inline pay-api/src/pay_api/services/base_payment_system.py:155]: I think the second check could cripple the database, it's feature flagged though

File: jobs/payment-jobs/tasks/ejv_partner_distribution_task.py [hunk 4]
```diff
@@ -23,14 +23,13 @@
...
-from pay_api.models import EjvLink as EjvLinkModel
+from pay_api.models import EjvInvoiceLink as EjvInvoiceLinkModel
 from pay_api.models import FeeSchedule as FeeScheduleModel
 from pay_api.models import Invoice as InvoiceModel
 from pay_api.models import PaymentLineItem as PaymentLineItemModel
-from pay_api.models import RefundsPartial as RefundsPartialModel
 from pay_api.models import Receipt as ReceiptModel
 from pay_api.models import db
-from pay_api.utils.enums import DisbursementStatus, EjvFileType, EJVLinkType, InvoiceStatus, PaymentMethod
+from pay_api.utils.enums import DisbursementStatus, EjvFileType, InvoiceStatus, PaymentMethod
```
Comment (seeker25) [inline jobs/payment-jobs/tasks/ejv_partner_distribution_task.py:32]: Will fix this entire file under my new PR: 
https://github.com/bcgov/sbc-pay/pull/1637

**Review by Jxio — APPROVED** [review]

File: pay-api/src/pay_api/models/invoice.py [hunk 5]
```diff
@@ -102,7 +102,7 @@ class Invoice(Audit):  # pylint: disable=too-many-instance-attributes
     payment_date = db.Column(db.DateTime, nullable=True)
     # default overdue_date to the first of next month
     overdue_date = db.Column(db.DateTime, nullable=True,
-                             default=lambda: datetime.now(tz=timezone.utc) + relativedelta(months=1, day=1))
+                             default=lambda: datetime.now(tz=timezone.utc) + relativedelta(months=2, day=1))
```
Comment (seeker25) [inline pay-api/src/pay_api/models/invoice.py:105]: Need migration to fix this...

**Review by ochiu — APPROVED** [review]

File: pay-api/gunicorn_config.py [hunk 6]
```diff
@@ -18,10 +18,10 @@
 import os
 
 # https://docs.gunicorn.org/en/stable/settings.html#workers
-workers = int(os.environ.get('GUNICORN_PROCESSES', '3'))  # gunicorn default - 1
+workers = int(os.environ.get('GUNICORN_PROCESSES', '1'))  # gunicorn default - 1
```
Comment (seeker25) [inline pay-api/gunicorn_config.py:21]: Should fix the issue where 3 migrators try to upgrade the database at the same time

**Pull request merged**